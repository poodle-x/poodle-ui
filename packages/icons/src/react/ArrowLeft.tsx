// This file is generated from svg file.
import React from "react";
import createSvgIcon from "../utils/createSvgIcon";

const ArrowLeft = createSvgIcon(
	<React.Fragment>
		<path d="m8 12 6 6.928V5.072L8 12z" />
	</React.Fragment>,
	"ArrowLeft"
);

export default ArrowLeft;
