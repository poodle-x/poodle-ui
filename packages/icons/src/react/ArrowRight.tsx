// This file is generated from svg file.
import React from "react";
import createSvgIcon from "../utils/createSvgIcon";

const ArrowRight = createSvgIcon(
	<React.Fragment>
		<path d="m16 12-6-6.928v13.856L16 12z" />
	</React.Fragment>,
	"ArrowRight"
);

export default ArrowRight;
