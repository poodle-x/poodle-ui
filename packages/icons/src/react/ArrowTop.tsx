// This file is generated from svg file.
import React from "react";
import createSvgIcon from "../utils/createSvgIcon";

const ArrowTop = createSvgIcon(
	<React.Fragment>
		<path d="m12 8-6.928 6h13.856L12 8z" />
	</React.Fragment>,
	"ArrowTop"
);

export default ArrowTop;
