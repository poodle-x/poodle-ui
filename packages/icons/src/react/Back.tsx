// This file is generated from svg file.
import React from "react";
import createSvgIcon from "../utils/createSvgIcon";

const Back = createSvgIcon(
	<React.Fragment>
		<path
			fillRule="evenodd"
			clipRule="evenodd"
			d="m8.026 6.335-4.701 4.856a1.171 1.171 0 0 0 0 1.618l.005.006 4.696 4.85a1.084 1.084 0 0 0 1.568 0 1.171 1.171 0 0 0 0-1.619l-2.81-2.902h13.109c.612 0 1.108-.512 1.108-1.144 0-.632-.496-1.145-1.108-1.145H6.783l2.81-2.901a1.171 1.171 0 0 0 0-1.619 1.084 1.084 0 0 0-1.567 0Z"
		/>
	</React.Fragment>,
	"Back"
);

export default Back;
