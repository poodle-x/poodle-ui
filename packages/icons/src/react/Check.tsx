// This file is generated from svg file.
import React from "react";
import createSvgIcon from "../utils/createSvgIcon";

const Check = createSvgIcon(
	<React.Fragment>
		<path d="M5.636 14.121a1 1 0 1 0-1.414 1.415l4.95 4.95L19.777 9.878a1 1 0 0 0-1.414-1.415L9.17 17.657 5.636 14.12z" />
	</React.Fragment>,
	"Check"
);

export default Check;
