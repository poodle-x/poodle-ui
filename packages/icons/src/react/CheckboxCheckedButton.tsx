// This file is generated from svg file.
import React from "react";
import createSvgIcon from "../utils/createSvgIcon";

const CheckboxCheckedButton = createSvgIcon(
	<React.Fragment>
		<path
			fillRule="evenodd"
			clipRule="evenodd"
			d="M4 2a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V4a2 2 0 0 0-2-2H4zm16.182 1.818H3.818v16.364h16.364V3.818z"
		/>
		<path d="M7.227 12.591a.75.75 0 1 0-1.06 1.06l3.712 3.713 7.955-7.955a.75.75 0 0 0-1.06-1.06l-6.895 6.894-2.652-2.652z" />
	</React.Fragment>,
	"CheckboxCheckedButton"
);

export default CheckboxCheckedButton;
