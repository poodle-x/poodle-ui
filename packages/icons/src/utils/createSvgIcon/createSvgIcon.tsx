import React from "react";

export default function createSvgIcon(
	path: React.ReactNode,
	iconName: string,
	config: {
		defaultSvgProps?: React.SVGProps<SVGSVGElement>;
	} = {}
): React.NamedExoticComponent<
	React.PropsWithoutRef<React.SVGProps<SVGSVGElement>> &
		React.RefAttributes<SVGSVGElement>
> {
	const Icon: React.ForwardRefRenderFunction<
		SVGSVGElement,
		React.SVGProps<SVGSVGElement>
	> = (props, ref) => (
		<svg
			ref={ref}
			width="1em"
			height="1em"
			viewBox="0 0 24 24"
			fill="currentColor"
			{...config.defaultSvgProps}
			{...props}
		>
			{path}
		</svg>
	);

	Icon.displayName = `${iconName}Icon`;

	return React.memo(
		React.forwardRef<SVGSVGElement, React.SVGProps<SVGSVGElement>>(Icon)
	);
}
