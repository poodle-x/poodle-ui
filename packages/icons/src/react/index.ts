import ArrowDown from "./ArrowDown";
import ArrowLeft from "./ArrowLeft";
import ArrowRight from "./ArrowRight";
import ArrowTop from "./ArrowTop";
import Back from "./Back";
import Check from "./Check";
import CheckboxButton from "./CheckboxButton";
import CheckboxCheckedButton from "./CheckboxCheckedButton";
import CheckboxIndeterminateButton from "./CheckboxIndeterminateButton";
import Clear from "./Clear";
import Delete from "./Delete";
import Document from "./Document";
import DocumentAdd from "./DocumentAdd";
import DocumentDelete from "./DocumentDelete";
import DocumentEdit from "./DocumentEdit";
import Edit from "./Edit";
import Error from "./Error";
import Exit from "./Exit";
import Folder from "./Folder";
import FolderCreate from "./FolderCreate";
import FolderDelete from "./FolderDelete";
import FolderEdit from "./FolderEdit";
import Menu from "./Menu";
import Plus from "./Plus";
import RadioButton from "./RadioButton";
import RadioCheckedButton from "./RadioCheckedButton";
import Search from "./Search";
import Settings from "./Settings";
import Trash from "./Trash";
import Warn from "./Warn";
import ZoomIn from "./ZoomIn";
import ZoomOut from "./ZoomOut";

export {
	ArrowDown,
	ArrowLeft,
	ArrowRight,
	ArrowTop,
	Back,
	Check,
	CheckboxButton,
	CheckboxCheckedButton,
	CheckboxIndeterminateButton,
	Clear,
	Delete,
	Document,
	DocumentAdd,
	DocumentDelete,
	DocumentEdit,
	Edit,
	Error,
	Exit,
	Folder,
	FolderCreate,
	FolderDelete,
	FolderEdit,
	Menu,
	Plus,
	RadioButton,
	RadioCheckedButton,
	Search,
	Settings,
	Trash,
	Warn,
	ZoomIn,
	ZoomOut,
};
