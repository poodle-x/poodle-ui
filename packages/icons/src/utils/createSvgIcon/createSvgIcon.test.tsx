import { render } from "@testing-library/react";
import React from "react";
import createSvgIcon from "./";

describe("createSvgIcon", () => {
	it("should return component", () => {
		const Component = createSvgIcon(
			<rect
				x="1"
				y="1"
				width="20"
				height="11"
				stroke="#282828"
				strokeWidth="2"
			/>,
			"Test",
			{
				defaultSvgProps: {
					clipPath: "1",
				},
			}
		);

		const { container } = render(<Component aria-busy="true" />);

		expect(container.firstChild).toMatchSnapshot();
	});
});
