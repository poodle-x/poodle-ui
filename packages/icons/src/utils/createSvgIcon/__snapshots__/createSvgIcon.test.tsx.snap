// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSvgIcon should return component 1`] = `
<svg
  aria-busy="true"
  clip-path="1"
  fill="currentColor"
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
>
  <rect
    height="11"
    stroke="#282828"
    stroke-width="2"
    width="20"
    x="1"
    y="1"
  />
</svg>
`;
